p2p/protocol/circuitv2/relay/metrics.go

```diff
@@ -0,0 +1,262 @@
+		prometheus.CounterOpts{
+			Namespace: metricNamespace,
+			Name:      "reservation_rejected_total",
+			Help:      "Relay Reservation Rejected Reason",
+		},
+		[]string{"reason"},
+	)
+
+	connectionTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: metricNamespace,
+			Name:      "connection_total",
+			Help:      "Relay Connection Total",
+		},
+		[]string{"type"},
+	)
+	connectionRequestStatusTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: metricNamespace,
+			Name:      "connection_request_status_total",
+			Help:      "Relay Connection Request Status",
+		},
+		[]string{"status"},
+	)
+	connectionRejectionTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: metricNamespace,
+			Name:      "connection_rejected_total",
+			Help:      "Relay Connection Rejected Reason",
+		},
+		[]string{"reason"},
+	)
+	connectionDurationSeconds = prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Namespace: metricNamespace,
+			Name:      "connection_duration_seconds",
+			Help:      "Relay Connection Duration",
+		},
+	)
+
+	bytesTransferredTotal = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: metricNamespace,
+			Name:      "bytes_transferred_total",
+			Help:      "Bytes Transferred Total",
+		},
+	)
+
+	collectors = []prometheus.Collector{
+		status,
+		reservationTotal,
+		reservationRequestStatusTotal,
+		reservationRejectedTotal,
+		connectionTotal,
+		connectionRequestStatusTotal,
+		connectionRejectionTotal,
+		connectionDurationSeconds,
+		bytesTransferredTotal,
+	}
+)
+
+const (
+	requestStatusOK       = "ok"
+	requestStatusRejected = "rejected"
+	requestStatusError    = "error"
+)
+
+const (
+	typeReceived = "received"
+	typeOpened   = "opened"
+	typeClosed   = "closed"
+)
+
+const (
+	rejectionReasonAttemptOverRelay      = "attempt over relay"
+	rejectionReasonDisallowed            = "disallowed"
+	rejectionReasonIPConstraintViolation = "ip constraint violation"
+	rejectionReasonResourceLimitExceeded = "resource limit exceeded"
+	rejectionReasonBadRequest            = "bad request"
+	rejectionReasonNoReservation         = "no reservation"
+	rejectionReasonClosed                = "closed"
+)
+
+// MetricsTracer is the interface for tracking metrics for relay service
+type MetricsTracer interface {
+	// RelayStatus tracks whether the service is currently active
+	RelayStatus(enabled bool)
+
+	// ConnectionRequestReceived tracks a new relay connect request
+	ConnectionRequestReceived()
+	// ConnectionOpened tracks metrics on opening a relay connection
+	ConnectionOpened()
+	// ConnectionClosed tracks metrics on closing a relay connection
+	ConnectionClosed(d time.Duration)
+	// ConnectionRequestHandled tracks metrics on handling a relay connection request
+	// rejectionReason is ignored for status other than `requestStatusRejected`
+	ConnectionRequestHandled(status string, rejectionReason string)
```

Comment:
Would it make sense to split this up in `ConnectionRequestHandled` (for success) and `ConnectionRequestRejected`? What would be the difference between `ConnectionRequestReceived` and `ConnectionRequestHandled`?

Reply:
yeah received was the same. I've removed ```ConnectionRequestReceived``` and ```ReservationRequestReceived``` and kept ```ReservationRequestHandled``` and ```ConnectionRequestHandled```